api.planx.uk/editor/publish.ts

```diff
@@ -241,27 +251,40 @@ const validateInviteToPay = (flow: Record<string, any>): ValidationResponse => {
   };
 };
 
-const inviteToPayEnabled = (flow: Record<string, any>): boolean => {
-  const payNodeStatuses = Object.entries(flow)
-    .filter(([_nodeId, nodeData]) => nodeData?.type === ComponentType.Pay)
-    ?.map(([_nodeId, nodeData]) => nodeData?.data?.allowInviteToPay);
+const inviteToPayEnabled = (flowGraph: FlowGraph): boolean => {
+  const payNodes = Object.entries(flowGraph).filter(
+    (entry): entry is [string, Node] =>
+      isComponentType(entry, ComponentType.Pay),
+  );
+  const payNodeStatuses = payNodes.map(
+    ([_nodeId, node]) => node?.data?.allowInviteToPay,
+  );
   return (
     payNodeStatuses.length > 0 &&
     payNodeStatuses.every((status) => status === true)
   );
 };
 
+const isComponentType = (
+  entry: Entry<FlowGraph>,
+  type: ComponentType,
+): entry is [string, Node] => {
+  const [nodeId, node] = entry;
+  if (nodeId === "_root") return false;
```

Comment:
An alternative (and possibly better) solution to this would be to add type property to the root node (`ComponentType._Root`?).